### PR TITLE
implemented changes to reflect issue @35

### PR DESCRIPTION
**Pull request by dgolman**
implemented fallback_text for slack integration and went ahead and did the same for hip_chat as well.

I cleaned up some of the tests in the spec/lib/integrations/slack_spec.rb to reflect the new changes I made to the lib/integrations/slack.rb

### CODE REVIEW

**Comment by dgolman** [comment]
#35

**Comment by ukd1** [comment]
@epaulet - can you review this? @dgolman is interviewing with us and wrote some code for this!

File: lib/integrations/hip_chat.rb [hunk 1]
```diff
@@ -54,6 +54,11 @@ def message_color
       color_hash[event_type]
     end
 
+    def self.message_text
```
Comment (epaulet) [inline lib/integrations/hip_chat.rb]: This looks like a class method to me. Do the specs test for this?

File: lib/integrations/message_formatter.rb [hunk 2]
```diff
@@ -31,6 +31,10 @@ def run_test_failure_message
       "has a failed a test!"
     end
 
+    def fallback_text
+      "retreat!"
```
Comment (epaulet) [inline lib/integrations/message_formatter.rb]: The fallback text should be the same message but with a text-only options (no markup/hyperlinks of any sort). This was not stated in the issue and I apologize :)

**Comment by epaulet** [comment]
@dgolman So I didn't put a lot of details in the GH issue because I thought I was going to do this myself, so you didn't have very much to work off of. The fallback text is what's described here: https://api.slack.com/docs/attachments

Just a text-only alternative to formatted text for clients who can't consume anything but plain text.

I'll update the issue with more details as well.

File: lib/integrations/hip_chat.rb (hunk 1 above)
Comment (dgolman) [inline lib/integrations/hip_chat.rb]: Great point. Added that method without any tests. Will take care of this.

File: lib/integrations/message_formatter.rb (hunk 2 above)
Comment (dgolman) [inline lib/integrations/message_formatter.rb]: No worries. I will refer to your comment below.

**Comment by dgolman** [comment]
@epaulet How would I test to make sure the fallback text is there on the spec side?

**Comment by epaulet** [comment]
I was thinking a test verifying that any json sent to the Slack API has the fallback field and it's not empty and doesn't contain any markdown.

File: lib/integrations/slack.rb [hunk 3]
```diff
@@ -31,93 +29,138 @@ def send_event
...
+        { title: "Browser", value: payload[:browser][:description], short: true }
+      ]
+    }
   end
 
   def webhook_timeout_fields
-    [{ title: "Environment", value: run[:environment][:name], short: false }]
+    {
+      fallback: message_text(fallback: true),
+      text: message_text,
```
Comment (epaulet) [inline lib/integrations/slack.rb]: this is kind of a nitpick, but every method contains these two lines. is there a way to keep this DRYer?

File: spec/lib/integrations/slack_spec.rb [hunk 4]
```diff
@@ -175,65 +175,4 @@
       end
     end
   end
-
-  describe '#message_color' do
```
Comment (epaulet) [inline spec/lib/integrations/slack_spec.rb]: Can you add specs test the color is correct for the new way you are determining it?

Comment (dgolman) [inline spec/lib/integrations/slack_spec.rb]: Yup, I will push the code along with the above fixes to make the two lines DRY.

File: lib/integrations/slack.rb [hunk 5]
```diff
@@ -31,21 +29,21 @@ def send_event
   private
 
   def attachments
-    attachment = {
-      text: message_text,
-      color: message_color
-    }
-
     case event_type
```
Comment (epaulet) [inline lib/integrations/slack.rb]: this is good, but for succinct and DRY code, you can do something like this.
```attachment = case event_type
when 'run_completion' then run_completion_fields
...```

**Comment by epaulet** [comment]
looks good to me other than the one comment.

File: lib/integrations/slack.rb (hunk 5 above)
Comment (dgolman) [inline lib/integrations/slack.rb]: I think I did this but it was pushed to the other pull request :/

**Pull request merged**